packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -125,6 +126,10 @@ class TransferEngine implements ITransferEngine {
     }
   }
 
+  async prepareTransfer(): Promise<void> {
+    await this.destinationProvider.prepareForStreaming?.();
+  }
```

Comment:
I'm starting to see all the provider methods as essentially hooks, especially since they are all optional (at least in the types, and in my test branch I've fixed them to be called optionally as well). So what I mean is, would it make sense to call this "beforeTransfer" and add it to the sourceProvider as well?

Reply:
yes I was thinking the same thing, beforeTransfer and beforeStreaming sound like good names.